packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -48,30 +50,53 @@ const enum EnvNodeTypes {
     COMMENT = 8,
 }
 
-function hydrate(vnode: VNode, node: Node) {
+let hasMismatch = false;
+
+export function hydrateRoot(vm: VM) {
+    hasMismatch = false;
+
+    runConnectedCallback(vm);
+    hydrateVM(vm);
+
+    if (hasMismatch) {
+        logError('Hydration completed with errors.', vm);
```

Comment:
Is there an error message we could put in here that would help the developer solve the mismatch?

Reply:
this is a general error shown after hydration is completed, on top of the errors shown to help devs to solve the mismatches.